### ISSUE TITLE
Code comment of `\texdimenwithunit` about some `E` possibly `1.0`, not always `0.ddd...` seems wrong

### ISSUE THREAD

**Issue by jfbu** [jfbu/texdimens#22]
For the `second argument<1pt` branch (and also  `=1pt` as it got merged into it), a certain quantity `C = ceil(r * 65536/f)` is obtained (indirectly) from which a decimal  `E` is then derived via `E pt = C sp`. Here `0<=r<f<=65536` and the code comments seem to leave open the possibility that `E=1.0`.

This seems wrong: `E=1.0` means  `C=65536`, i.e. `r*65536/f>65535` i.e. `r>f - f/65536`. But `f-1>=r` so we get `f-1>f-f/65536` i.e. `f/65536>1`. And this is false.

Am I correct? If yes, please amend your too hastily written code comments please.

**Comment by RuixiZhang42** [comment]
From the ceiling, `C` can only go up to 65535, yes. But there is a later step that possibly add one to `C`, then `E` would be `1.0`. I didn’t have the time to rule out this `C<-C+1` case.

**Comment by jfbu** [comment]
> From the ceiling, `C` can only go up to 65535, yes. But there is a later step that possibly add one to `C`, then `E` would be `1.0`. I didn’t have the time to rule out this `C<-C+1` case.

It is not `C<-C+1` but `C<-R+1` to "fix" `R` in order to get `C`. (`R=round(r*65536/f)` can be evaluated directly, `C=ceil(r*65536/f)` is not as easy).

The code comment in 1.0 release:
https://github.com/jfbu/texdimens/blob/36d8833c6d72d6d51d2e7e1e652e10dfcf183931/texdimens/texdimens.tex#L873-L877
is wrong to say the  final `E` can be `1.0` "now that we use the ceil approach in this branch". When I wrote it I probably made quick wrong reasoning "I know that `R<65536`, so I can only say `C<= 65536` for ceiling",  I did not re-consider the formula for `C` itself from which one can see that in fact `C` is `65535` at most. Now fixed at #23, for 1.1 release, as part of re-writing the manner to check if one must do `C<-R` or `C<-R+1`.
